features/atcryptoauth/ATCADevice.h

```diff
@@ -0,0 +1,283 @@
+     */
+    ATCAKey * GetKeyToken(ATCAKeyID keyId, ATCAError & err);
+
+    /** Dump config zone registers on standard output.
+     */
+    void      DumpConfig();
+
+private:
+    /** Perform device wakeup procedure.
+     *
+     * Wakeup procedure consists of:
+     *  - Wait Tpu for device Up
+     *  - Pull SDA low for Twlo
+     *  - SDA back high
+     *  - Wait Twhi
+     *  - Ready. A following read should get status 0x11(Wake token received)
+     */
+    ATCAError   Wakeup();
+    /** Reset watchdog timer.
+     *
+     *  Avoid deep sleep and loss of current command state on watchdog timer
+     *  expiry by putting device in Idle mode and straight away waking up.
+     */
+    ATCAError   ResetWatchdog();
+
+    /** Execute a command.
+     *
+     *  This function performs command execution based on datasheet section 9.1:
+     *      - Constructs command tx buffer based on device IO protocol.
+     *      - Transmit tx buffer.
+     *      - Receive response.
+     *      - Check response status.
+     *      - Recover from watchdog expiry or sync failure.
+     *      - Returns response and status back to caller.
+     *
+     *  @param info Command parameters
+     *  @return     Error code from enum ATCAError.
+     */
+    ATCAError   RunCommand(ATCACmdInfo * info);
+
+    /** Execute a read command. This function limits the read data to 4 bytes.
+     *
+     *  Byte address is required to be 4 byte aligned.
+     *
+     *  @param zone     Device EEPROM zone.
+     *  @param address  Byte address within a zone.
+     *  @param word     output buffer of 4 bytes.
+     *  @param wlen     Output buffer length.
+     *  @return         Error code from enum ATCAError.
+     */
+    ATCAError   ReadCommand(ATCAZone zone, uint16_t address,
+                            uint8_t * word, size_t wlen);
+
+    /** Execute a write command. Writes are also limited to 4 bytes.
+     *
+     *  Byte address is required to be 4 byte aligned.
+     *
+     *  @param zone     Device EEPROM zone.
+     *  @param address  Byte address within a zone.
+     *  @param word     input buffer of 4 bytes.
+     *  @param wlen     input buffer length.
+     *  @return         Error code from enum ATCAError.
+     */
+    ATCAError   WriteCommand(ATCAZone zone, uint16_t address,
+                             const uint8_t * word, size_t wlen);
+
+    /** Execute a Nonce command.
+     *
+     *  It loads data into Device Tempkey that is used as input in subsequent
+     *  Sign or Verify commands.
+     *
+     *  @param message  Data(generally message digest) to load in Tempkey.
+     *  @param len      message length.
+     *  @return         Error code from enum ATCAError.
+     */
+    ATCAError   Nonce(const uint8_t * message, size_t len);
+
+    /** Read 4 bytes from configuration zone.
+     *
+     *  @param address  Byte address in Data zone. Should be 4 byte aligned.
+     *  @param len      output buffer length.
```

Comment:
Please state explicitly that it's the length of `obuf`, and state the constraint on the length. This parameter has to be exactly 4, right?